src/main/java/io/appium/java_client/pagefactory/bys/builder/Strategies.java

```diff
@@ -39,6 +39,16 @@ enum Strategies {
             return super.getBy(annotation);
         }
     },
+    BYTESTTAG("composeTestTag") {
+        @Override By getBy(Annotation annotation) {
+            String value = getValue(annotation, this);
+            if (annotation.annotationType().equals(AndroidFindBy.class)
+                    || annotation.annotationType().equals(AndroidBy.class)) {
+                return AppiumBy.androidUIAutomator("new UiSelector().resourceId(\"" + value + "\")");
```

Comment:
is it needed to just create a synonym to the existing location strategy?

Reply:
yes, but it reduce possibility of errors in the locators, as a result code with annotation became simpler and clear.
Compose is very common approach now days and is recommended by Google, I hope this small change will make writing tests for Compose a bit simpler.

Reply:
https://github.com/appium/appium/issues/18081

Reply:
Looks like in provided issue there is the same problem that I try to solve.
I found great article, where this problem is well described:
https://medium.com/@pourzand.p/automating-android-jetpack-compose-app-with-appium-70aa7a5f80e2

Reply:
I don't observe the above article mentioning the usage of the disableIdLocatorAutocompletion setting. Have you actually tried it?

Reply:
Yes, I tried and it works well in case when the project contains only Compose views. In my project I have both types of views, and when I set disableIdLocatorAutocompletion = true Compose testTags work, but regular view ids don't work (because regular android views always have package in the id).

Reply:
that is why it was done as a setting to be able to switch the mode dynamically during the session lifecycle